addons/payment/views/payment_provider_views.xml

```diff
@@ -63,9 +65,9 @@
                         Please use the <em>Duplicate</em> action instead.
                     </div>
                     <group>
-                        <group name="payment_state">
-                            <field name="code" groups="base.group_no_one" attrs="{'readonly': [('id', '!=', False)], 'invisible': [('module_id', '!=', False), ('module_state', '!=', 'installed')]}"/>
-                            <field name="state" widget="radio" attrs="{'invisible': [('module_state', '=', 'uninstalled')]}"/>
+                        <group name="payment_state" attrs="{'invisible': [('module_state', '!=', 'installed')]}">
```

Comment:
Just a small detail: when the provider is not created yet, the field `code` should still be visible so the user can change it before saving. Indeed, after saving, the field is readonly. The field is invisible because the module is not (and will never be) set.

To fix that, you must change the `attrs` of `<group name="payment_state">` to `attrs="{'invisible': [('module_state', 'not in', '(installed, False)')]}"`.

Reply:
hi @AntoineVDV ,
updated the as per the suggestion and changed the commit message.
thanks